lib/versioncake/view_additions.rb

```diff
@@ -5,87 +5,237 @@
+      locales = "[a-z]{2}(?:-[A-Z]{2})?"
+      variants = "[^.]*"
+
+      %r{
+          \A
+          (?:(?<prefix>.*)/)?
+          (?<partial>_)?
+          (?<action>.*?)
+          (?:\.(?<locale>#{locales}))??
+          (?:\.(?<format>#{formats}))??
+          (?:\.(?<versions>v[0-9]+))??
+          (?:\+(?<variant>#{variants}))??
+          (?:\.(?<handler>#{handlers}))?
+          \z
+        }x
+    end
+
+    def parse(path)
+      @regex ||= build_path_regex
+      match = @regex.match(path)
+      path = ActionView::TemplatePath.build(match[:action], match[:prefix] || "", !!match[:partial])
+      details = ActionView::TemplateDetails.new(
+        match[:locale]&.to_sym,
+        match[:handler]&.to_sym,
+        match[:format]&.to_sym,
+        match[:variant]&.to_sym,
+        match[:versions]&.to_sym,
+      )
+
+      ActionView::Resolver::PathParser::ParsedPath.new(path, details)
     end
   end
 
-  # The default extract handler expects that the handler is the last extension and
-  # the format is the next one. Since we are replacing the DEFAULT_PATTERN, we need to
-  # make sure that we extract the format from the correct position.
-  #
-  # The version may be stuck inbetween the format and the handler. This is actually pretty tricky
-  # because the version is optional and the locale is optional-which means when there are 3 'pieces'
-  # it may be the locale, format and handler or the format, version and handler. To check this, we will
-  # try one additional time if there are more pieces, which should cover all the cases:
-  #
-  # Cases:
-  #  1: assume version is in the extension, pieces = ['html','erb']
-  #  2: assume version is in the extension, pieces = ['html','v1','erb']
-  #  3: assume version is in the extension, pieces = ['en','html','erb']
-  #  4: assume version is in the extension, pieces = ['en','html','v1','erb']
-  #
-  def extract_handler_and_format(path, default_formats)
-    pieces = File.basename(path).split(".")
-    pieces.shift
-
-    extension = pieces.pop
-    if ActionPack::VERSION::MAJOR == 4
-      unless extension
-        message = "The file #{path} did not specify a template handler. The default is currently ERB, " \
-                    "but will change to RAW in the future."
-        ActiveSupport::Deprecation.warn message
+  ActionView::TemplateDetails::Requested.class_eval do
+    attr_reader :locale, :handlers, :formats, :variants, :versions
+    attr_reader :locale_idx, :handlers_idx, :formats_idx, :variants_idx, :versions_idx
+
+    def initialize(locale:, handlers:, formats:, variants:, versions:)
+      @locale = locale
+      @handlers = handlers
+      @formats = formats
+      @variants = variants
+      @versions = versions
+
+      @locale_idx   = build_idx_hash(locale)
+      @handlers_idx = build_idx_hash(handlers)
+      @formats_idx  = build_idx_hash(formats)
+      @versions_idx  = build_idx_hash(versions)
+      if variants == :any
+        @variants_idx = ANY_HASH
```

Comment:
Yeah, I noticed this too in other places. It seems that referencing any class/constant in class_eval doesn't have the same context as referencing it within the class itself. The way I was fixing it was to add the namespace here explicitly. My test cases probably didn't exercise this branch, so that's why I didn't run into it yet.

I can update this to add the namespace.

Reply:
that fixed it for me thanks 👍🏻